lib/improver/tests/utilities/test_TemporalInterpolation.py

```diff
@@ -158,77 +158,92 @@ def test_time_list_out_of_bounds(self):
         data_time_0 = np.ones((self.npoints, self.npoints), dtype=np.float32)
         cube_time_0 = set_up_variable_cube(data_time_0,
                                            time=time_start,
                                            frt=time_start)
-        cube_times = add_coordinate(cube_time_0,
+        cube_times = add_coordinate(cube_time_0.copy(),
                                     [time_start, time_mid, time_end],
                                     'time', is_datetime=True)
-        self.cube = add_coordinate(cube_times, self.extra_coord,
-                                   'extra_coord')
-        self.bad_coords = self.cube.copy()
-        self.bad_coords.coord('time').points = (
-            self.bad_coords.coord('time').points.astype(np.float64))
-        self.bad_coords.coord('forecast_period').points = (
-            self.bad_coords.coord('forecast_period').points.astype(np.int64))
-        self.bad_coords.coord('extra_coord').points = (
-            self.bad_extra_coord)
-
-    def test_check_all_type(self):
-        """Test that a cube is returned.with the right types"""
+        # Convert units and datatypes, so that they are non-standard.
+        cube_times.coord("time").convert_units(
+            "hours since 1970-01-01 00:00:00")
+        cube_times.coord("time").points = (
+            cube_times.coord("time").points.astype(np.int32))
+        cube_times.coord("forecast_reference_time").convert_units(
+            "hours since 1970-01-01 00:00:00")
+        cube_times.coord("forecast_reference_time").points = (
+            cube_times.coord(
+                "forecast_reference_time").points.astype(np.int32))
+        cube_times.coord("forecast_period").convert_units("hours")
+        cube_times.coord("forecast_period").points.astype(np.float32)
+
+        self.cube = cube_times
+
+    def test_check_points(self):
+        """Test that a cube is returned with the right types"""
         plugin = TemporalInterpolation(interval_in_minutes=60)
-        result = plugin.check_cube_coords_dtype(self.cube,
-                                                self.bad_coords)
+        result = plugin.enforce_time_coords_dtype(self.cube)
         self.assertIsInstance(result, iris.cube.Cube)
-        # All coords reverted back to the dtypes of the
-        # original cube and points in extra_coord cube have been rounded
+        # All coordinates converted to the desired units and datatypes.
+        # Check time coordinate.
         self.assertEqual(result.coord('time'),
                          self.cube.coord('time'))
         self.assertEqual(str(result.coord('time').points.dtype),
                          'int64')
+        # Check forecast_reference_time.
+        self.assertEqual(result.coord('forecast_reference_time'),
+                         self.cube.coord('forecast_reference_time'))
+        self.assertEqual(
+            str(result.coord('forecast_reference_time').points.dtype), 'int64')
```

Comment:
Although this does actually test something you never cast the coordinates away from int64 on the inputs, so this will always evaluate as true.

Reply:
The forecast_reference_time coordinate is converted to int32 within the setUp, so the test should be converting this to int64, which is the intended datatype.

Reply:
OK fair, I missed that.